src/HDF5.jl

```diff
@@ -1387,6 +1387,14 @@ function read(obj::DatasetOrAttribute, ::Type{T}, I...) where T
               sizeof($memtype) = $(h5t_get_size(memtype.id))
               """)
     end
+    if h5t_get_class(memtype.id) == HDF5.H5T_STRING
```

Comment:
I'm not sure that these checks are a good idea here. I envisioned this function as being somewhat unsafe and allowing you to read into basically any datatype as long as the sizes match. I think having this kind of interface which is 'high level' in some sense but still lets you do what you want is useful to have. For example I could imagine wanting to read data from a fixed string field into something other than a `FixedString`.

Reply:
Sure, I'd be fine dropping this check specifically. I only noticed it because `"variable"` happens to be 8 bytes, so it make it past the size check when you try to read the variable pointer as a fixed string. Just thought I'd point it out and see what other had to say about it — I can easily delete this and the corresponding tests.